.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+# Standard hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+
+# Python files
+-   repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+    -   id: black
+-   repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.0.270
+    hooks:
+    -   id: ruff
```

Comment:
Since we are starting this repository from scratch, it would be nice to use `ruff` to its full potential. We'll be able to get rid of the `black` `pre-commit` hook with `--fix` - https://github.com/scikit-hep/vector/blob/main/.pre-commit-config.yaml#L26-L31

Also, we can enable a ton of extensions (I don't know if they are called extensions or something else) - https://github.com/scikit-hep/vector/blob/main/pyproject.toml#L199-L228.

You can generally use https://github.com/scikit-hep/vector/blob/main/.pre-commit-config.yaml as a reference for all the pre-commit stuff. There are some doc specific pre-commits included in that file too.

Reply:
I thought of using the extra options that `ruff` provides, but I chose to skip them for now because this repository will rarely have any Python files to lint / auto-format – most files are static. I'll make some changes to remove `black`

Reply:
Ah, makes sense to leave it as it is then!